### PR TITLE
Vendor dependencies, update k8s to 1.16.3

### PR DESCRIPTION
**Pull request by sslavic**
Towards giantswarm/giantswarm#7675

### CODE REVIEW

**Comment by tomahawk28** [comment]
We were getting an error *before RBAC changed*
```
k logs -ndraughtsman deploy/draughtsman | luigi
D 11/29 16:23:17 creating in-cluster REST config | k8sclient/k8srestconfig/k8s_rest_config.go:148
D 11/29 16:23:17 created in-cluster REST config | k8sclient/k8srestconfig/k8s_rest_config.go:155
D 11/29 16:23:17 checking connection to Kubernetes | draughtsman/service/configurer/configmap/configmap.go:69
D 11/29 16:23:17 checking connection to Kubernetes | draughtsman/service/configurer/secret/secret.go:69
D 11/29 16:23:17 logging into registry | draughtsman/service/installer/helm/helm.go:193 | registry=quay.io | username=giantswarm+ghost
D 11/29 16:23:17 running helm command | draughtsman/service/installer/helm/helm.go:163 | name=login
D 11/29 16:23:26 ran helm command | draughtsman/service/installer/helm/helm.go:175 | name=login | stderr= | stdout= >>> Login succeeded
D 11/29 16:23:26 checking connection to Slack | draughtsman/service/notifier/slack/slack.go:86
D 11/29 16:23:28 finding if tiller is installed in namespace `kube-system` | helmclient/ensure_tiller_installed.go:37
D 11/29 16:23:28 running server at http://0.0.0.0:8000 | microkit/server/server.go:308
D 11/29 16:23:28 found that tiller is not installed in namespace `kube-system` | helmclient/ensure_tiller_installed.go:43
D 11/29 16:23:28 creating serviceaccount `tiller-giantswarm` in namespace `kube-system` | helmclient/ensure_tiller_installed.go:58
D 11/29 16:23:28 serviceaccount `tiller-giantswarm` in namespace `kube-system` already exists | helmclient/ensure_tiller_installed.go:73
D 11/29 16:23:28 creating clusterrolebinding `tiller-kube-system` | helmclient/ensure_tiller_installed.go:91
D 11/29 16:23:28 clusterrolebinding `tiller-kube-system` already exists | helmclient/ensure_tiller_installed.go:117
D 11/29 16:23:28 creating clusterrole `tiller-giantswarm-psp` | helmclient/ensure_tiller_installed.go:130
panic: [{/go/src/github.com/giantswarm/draughtsman/service/service.go:174: service.Boot} {/go/src/github.com/giantswarm/draughtsman/vendor/github.com/giantswarm/helmclient/ensure_tiller_installed.go:163: } {clusterroles.rbac.authorization.k8s.io is forbidden: User "system:serviceaccount:draughtsman:draughtsman" cannot create resource "clusterroles" in API group "rbac.authorization.k8s.io" at the cluster scope}]
goroutine 70 [running]:
github.com/giantswarm/draughtsman/service.(*Service).Boot(0xc0008320f0, 0x1bf2280, 0xc000054008, 0xc0006d52e0, 0x1bcddc0)
	/go/src/github.com/giantswarm/draughtsman/service/service.go:174 +0x15d
created by main.mainError.func1
	/go/src/github.com/giantswarm/draughtsman/main.go:93 +0x3a0
```
@rossf7 suggested it may happen since k8s 1.16.3 had moved `PodSecrityPolicy` object from `extensions/v1beta` into `policy/v1beta1`.
But as I checked into our testing installation, it still using 1.15.4 at least in ginger. 

```
opsctl create kubeconfig -i ginger --user jhbok
New key-pair created with expiry of 1 day(s).

Switched to kubectl context 'giantswarm-ginger'. kubectl is set up. Check it out using:

    kubectl cluster-info

Whenever you want to switch to using this context do:

    kubectl config use-context giantswarm-ginger

 ~  k version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:49Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.4", GitCommit:"67d2fcf276fcd9cf743ad4be9a9ef5828adc082f", GitTreeState:"clean", BuildDate:"2019-09-18T14:41:55Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

Could only client version change affect retrieval of `PodSecurityPolicy` by resource deprecation even if the server version is the same?

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.15.md#deprecations-and-removals

**Comment by sslavic** [comment]
Thanks @tomahawk28  for testing draughtsman dependency upgrade! Unfortunately we do not have even basic e2e integration test for its chart.

> We were getting an error _before RBAC changed_

Are you saying that you were getting that error while testing k8s-1-16 branch changes and then https://github.com/giantswarm/draughtsman/commit/ae91ba24010a139c5788634ad08d6bbd6d02d549 fixed it?

ClusterRole `tiller-giantswarm-psp` creation is done in this k8s-1-16 branch https://github.com/giantswarm/draughtsman/blob/k8s-1-16/vendor/github.com/giantswarm/helmclient/ensure_tiller_installed.go#L158 only since it vendors latest helmclient with https://github.com/giantswarm/helmclient/pull/124 (/cc @glitchcrab) - that same helmclient change was not vendored to the master branch of draughtsman. Unfortunately we do not have monorepo, don't warn of outdated dependencies, so these kind of breaking changes in libraries get detected late, coupled with other changes so it's hard to detect underlying cause.

Anyway, I'd prefer that we cleanup k8s-1-16 branch to include only k8s+helm upgrade stuff i.e. next week we should:
- vendor helmclient master to draughtsman vendor-deps branch from latest master,
- redo `create` verb addition to draughtsman-chart rbac.yaml there,
- test draughtsman vendor-deps branch, and if all OK
- merge k8s-1-16 branch to draughtsman master, and finally
- rebase this k8s-1-16 branch on top of latest draughtsman master so it includes only k8s+helm lib upgrade.

Even though this issue you uncovered was not `extensions/v1beta` -> `policy/v1beta1` related, after seeing that helmclient code now I propose that we do additional work in helmclient first (so before steps listed above vendoring helmclient master in draughtsman master). Specifically that code creating ClusterRole `tiller-giantswarm-psp` it seems has to be PSP updated/migrated too, it has a rule associate that allows it to `use` PSP from `extensions` API group - we should migrate that usage to `policy` API group too, and this can be done to helmclient master already (`policy/v1beta1` is available since 1.10 https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/ ). Even though this change is a must only for 1.16 k8s clusters, since it's already possible to perform this PSP migration in isolation from rest of 1.16 client and server rollout, IMO we should do it. Sooner it gets migrated, less chance there will be for copy-paste reuse of code referencing deprecated APIs. All this was my rationale for minimizing k8s-1-16 branch in other projects too. Seems I wasn't thorough enough for helmclient, didn't find this usage of deprecated APIs in Go code.

**Comment by sslavic** [comment]
Just discovered https://github.com/giantswarm/chart-operator/pull/296 by @glitchcrab  - I'm working on helmclient PSP migration bit, then will rebase the PR in chart-operator but also create one in draughtsman.

**Comment by glitchcrab** [comment]
sorry @sslavic, AWS certs got in the way of getting that one tested and merged. Appreciate you picking this up :+1:

File: helm/draughtsman-chart/templates/rbac.yaml [hunk 1]
```diff
@@ -70,6 +70,13 @@ rules:
       - get
       - create
       - update
+  - apiGroups:
+      - policy
+    resources:
+      - podsecuritypolicies
+    verbs:
+      - create
+      - use
```
Comment (rossf7) [inline helm/draughtsman-chart/templates/rbac.yaml]: This is added so helmclient can create a PSP for tiller.

**Review by rossf7 — COMMENTED** [review]
Tested on gauss. 

Needs to be deployed at the same time as https://github.com/giantswarm/chart-operator/pull/325 as both components share the same tiller.

**Comment by tomahawk28** [comment]
We can enforce draughtsman not to update any tiller, so chart-operator can handle that, WDYT?
https://github.com/giantswarm/helmclient/pull/142

**Comment by rossf7** [comment]
> We can enforce draughtsman not to update any tiller

Currently there are only chart CRs for aws-operator on AWS CPs so that won't work. Once there are chart CRs in all CPs we can do that.

**Comment by rossf7** [comment]
Tested on geckon. Going with this.

**Pull request merged**